earth_enterprise/src/fusion/autoingest/StorageManager_unittest.cpp

```diff
@@ -170,6 +179,18 @@ TEST_F(StorageManagerTest, PurgeCache) {
   }
 }
 
+TEST_F(StorageManagerTest, PurgeCacheBasedOnMemoryUtilization) {
+  storageManager.SetCacheMemoryLimit(1, MEMORY_LIMIT);
+  size_t i;
+  for(i = 0; i < CACHE_SIZE + 2; ++i) {
+    stringstream s;
+    s << "asset" << i;
+    Get<TestHandle>(storageManager, s.str(), false, true, false);
+    ASSERT_EQ(storageManager.DirtySize(), 0) << "Storage manager has unexpected item in dirty map";
+  }
+  ASSERT_LE(storageManager.CacheSize(), i+1) << "Unexpected number of items in cache";
+}
+
```

Comment:
Thinking about some other tests we could add:
- Memory limit that is greater than 0
- Check that the value returned from StorageManager::CacheMemoryUse makes sense